### DISCUSSION TITLE
Uvicorn does not reload when changes on docker mounted volume in WSL

### DISCUSSION THREAD

**Discussion by yonatanramirez** [Kludex/uvicorn#1893] [Potential Issue] [answered]
I'm currently using Uvicorn through FastAPI.
I'm using Ubuntu on WSL2 in Windows 11.

I start my project using docker compose, latest version.

It seems there is an issue detecting the files that changed on a mounted volume.

In my Dockerfile, I have set `ENV WATCHFILES_FORCE_POLLING=true`

The command used to launch Uvicorn is `uvicorn app.main:app --host 0.0.0.0 --port 8080 --reload --reload-dir /app/`

And my code is under ./src/backend/app, mounted to /app/ inside the container

What is weird, is that using watch('/app') directly from a python shell inside the container, the changes are logged correctly.

Here are some logs

**docker compose up logs**
```
[+] Running 2/2
 ⠿ Network quiz_default      Created                                    0.1s
 ⠿ Container quiz-backend-1  Created                                    0.0s
Attaching to quiz-backend-1
quiz-backend-1  | INFO:     Will watch for changes in these directories: ['/app']
quiz-backend-1  | INFO:     Uvicorn running on http://0.0.0.0:8080 (Press CTRL+C to quit)
quiz-backend-1  | INFO:     Started reloader process [1] using WatchFiles
quiz-backend-1  | INFO:     Started server process [8]
quiz-backend-1  | INFO:     Waiting for application startup.
quiz-backend-1  | INFO:     Application startup complete.
```

**docker compose exec backend python logs, run in parallel with above command**
```
Python 3.11.2 (main, Mar  1 2023, 14:56:17) [GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from watchfiles import watch
>>> for changes in watch('/app'):
...      print(changes)
... 
{(<Change.modified: 2>, '/app/models/question.py')}
{(<Change.modified: 2>, '/app/models/question.py')}
{(<Change.modified: 2>, '/app/api/api_v1/endpoints/question.py')}
{(<Change.modified: 2>, '/app/api/api_v1/endpoints/question.py')}
```

Any ideas if I'm missing something in the uvicorn command ?

I tried to launch programmatically, but the logs are the same
```
if __name__ == '__main__':
    uvicorn.run(
        "app.main:app",
        host="0.0.0.0",
        port=8080,
        reload=True,
        reload_dirs=['/app'],
    )

```

Thank you in advance for any idea

**Comment by euri10** [comment]
I'd try first without the reload_dirs kwarg

**Comment by yonatanramirez** [comment, marked as answer]
Hello,
I finally solved this.
The issue is that the working directory inside the Docker container was /, and it seems that reloading does not work in that case.

Changing the workdir to /app, setting PYTHONPATH to /, and launching with `uvicorn app.main:app --host 0.0.0.0 --port 8080 --reload` works as intended.

**Reply by rsurgiewicz** [reply]
kudos Mate! That saved my day!

**Reply by NormalNickYT** [reply]
Thanks <3
